react-components/src/architecture/base/utilities/extensions/arrayExtensions.ts

```diff
@@ -130,3 +130,20 @@ export function containsTheSameSet<T>(array1: readonly T[], array2: readonly T[]
   const set2 = new Set(array2);
   return every(array1, (element) => set2.has(element));
 }
+
+/**
+ * Counts the number of elements in the given array that satisfy the specified predicate.
+ *
+ * @param array - The array to search through.
+ * @param predicate - A function that tests each element for a condition.
+ * @returns The number of elements in the array that satisfy the predicate.
+ */
+export function count<T>(array: readonly T[], predicate: (t: T) => boolean): number {
+  let count = 0;
+  for (const item of array) {
```

Comment:
Should we write this to be defined in terms of the generatorsUtil-count-function (or write the generatorsUtil-count-function in terms of this function)?